### DISCUSSION TITLE
Make possible to pass context to <Outlet />

### DISCUSSION THREAD

**Discussion by tykhan** [TanStack/router#1153] [General]
react-router-dom outlet can take any context which will be passed to all sub-routes and it makes it really powerful feature. Would be great to implement something similar. Or there are any ways to achieve the same result?

`<Outlet context={{ randomValue: 1 }} />`

**Comment by hsunami10** [comment]
Def want something like this too ^

**Comment by alex-delia** [comment]
Would definitely be interested in seeing Tanner's or someone else from the Tanstack team's thoughts on implementing this.

**Comment by nidrizi** [comment]
I might be wrong but I think you can you the [Router Context](https://tanstack.com/router/latest/docs/framework/react/guide/router-context) to achieve the same thing
